src/H5Ppublic.h

```diff
@@ -357,11 +357,72 @@ typedef enum H5D_mpio_no_collective_cause_t {
     /**< Collective I/O was not performed because parallel filtered writes are disabled */
     H5D_MPIO_ERROR_WHILE_CHECKING_COLLECTIVE_POSSIBLE = 0x80,
     /**< Error */
-    H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE = 0x100
+    H5D_MPIO_NO_SELECTION_IO = 0x100,
+    /**< Collective I/O would be supported by selection or vector I/O but that feature was disabled
+       (see causes via H5Pget_no_selection_io_cause()) */
+    H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE = 0x200
     /**< Sentinel */
 } H5D_mpio_no_collective_cause_t;
 //! <!-- [H5D_mpio_no_collective_cause_t_snip] -->
 
+/**
```

Comment:
Why are these not in a struct like the other no*cause cases?

Reply:
The others are in enums. We decided to avoid putting bitwise values in enums because that is inconsistent with the normal usage of enums and potentially confusing, since the flags parameter can take on a value that does not match one of the enum values. In addition, unless I'm missing something there's no advantage using enum values instead of simple #defines.

Reply:
They are unsigned because they are bitflag values and need to be able to be combined in a bitwise fashion

Reply:
We plan to change the other no*cause values to definitions with 2.0